### DISCUSSION TITLE
Fly camera with zoom in iOS

### DISCUSSION THREAD

**Discussion by huyparody** [maplibre/maplibre-native#2718] [Q&A]
How do I set the zoom level to fly the camera at a specific zoom level? Currently, the behavior is fly to maximum zoom level using 

```swift
- (void)flyToCamera:(MLNMapCamera *)camera
         withDuration:(NSTimeInterval)duration
    completionHandler:(nullable void (^)(void))completion;
```

**Comment by louwers** [comment]
The flyTo method is specifically designed to zoom out and in to offer the user some context while moving the camera. https://www.win.tue.nl/~vanwijk/zoompan.pdf#page=5

Can you describe the effect you are looking for?

**Reply by huyparody** [reply]
> The flyTo method is specifically designed to zoom out and in to offer the user some context while moving the camera. https://www.win.tue.nl/~vanwijk/zoompan.pdf#page=5
> 
> 
> 
> Can you describe the effect you are looking for?

Im looking for a function works the same as easeTo with zoom level like MapBox, is that possible?

**Reply by louwers** [reply]
This is probably what you are looking for:

https://maplibre.org/maplibre-native/ios/latest/documentation/maplibre/mlnmapview/setcamera:withduration:animationtimingfunction:

**Reply by huyparody** [reply]
> This is probably what you are looking for:
> 
> https://maplibre.org/maplibre-native/ios/latest/documentation/maplibre/mlnmapview/setcamera:withduration:animationtimingfunction:

Oh yeah, the setCamera function is equivalent to MapBox and works fine with zoom level. Still, I want to find the equivalent of the [easeTo](https://docs.mapbox.com/ios/maps/examples/camera-animation/), for smoother camera animation, for some use cases like I want to fly from somewhere on the screen to the current user location, using setCamera is not a good practice.
